scala-generator/src/main/scala/models/Play2Json.scala

```diff
@@ -144,19 +144,39 @@ case class Play2Json(
   }
 
   private[this] def readersWithDiscriminator(union: ScalaUnion, discriminator: String): String = {
+    val defaultDiscriminatorTypeName: Option[String] = unionTypesWithNames(union).flatMap { case (t, typeName) =>
```

Comment:
this selects out the default union type name to use, if any was specified. It is acceptable for there to be no default (as is today).